app/scripts/modules/core/src/artifact/ArtifactIconService.ts

```diff
@@ -0,0 +1,29 @@
+const unknownArtifactPath = require('./icons/unknown-type-artifact.svg');
+
+interface IArtifactIcon {
+  type: RegExp;
+  path: string;
+}
+
+export class ArtifactIconService {
+  private static icons: IArtifactIcon[] = [];
+
+  public static registerType(type: RegExp, path: string) {
+    ArtifactIconService.icons.push({ type, path });
+  }
+
+  public static getPath(type: string) {
+    const icon = ArtifactIconService.icons.find(entry => entry.type.test(type));
+    if (icon == null) {
+      return unknownArtifactPath;
+    }
+    return icon.path;
+  }
+}
+
+ArtifactIconService.registerType(/docker\/image/, require('./icons/docker-image-artifact.svg'));
+ArtifactIconService.registerType(/kubernetes\/.*/, require('./icons/kubernetes-artifact.svg'));
+ArtifactIconService.registerType(/embedded\/base64/, require('./icons/embedded-base64-artifact.svg'));
+ArtifactIconService.registerType(/gcs\/object/, require('./icons/gcs-file-artifact.svg'));
+ArtifactIconService.registerType(/github\/file/, require('./icons/github-file-artifact.svg'));
+ArtifactIconService.registerType(/s3\/object/, require('./icons/s3-object-artifact.svg'));
```

Comment:
I'm torn about whether we should use regular expressions here vs. just an exact matching here.  It looks like we're only taking advantage of the regexes for kubernetes---how many different types of `kubernetes/` artifacts are there, and is it a small enough list we know a priori?

I also briefly wondered if it would make sense to just include the icon at the time we register the artifact itself (ex: in `gcs.artifact.ts` include a key for `artifact`) but it looks like not all artifact types are registered in that way, so maybe that is not an option.

Reply:
I don't know the exact number Spinnaker would have to be concerned with but there are definitely a lot of resource types.  The full list for 1.10 is here: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.10/.  I also think there's an element of not wanting to have to push code changes if the k8s project adds new kinds.